docs/Actions/apply-tokens.md

```diff
@@ -0,0 +1,165 @@
+
+## `Don't use it to`
+
+- Concatenate 2 tokens - Use [Inject Data](/docs/Actions/inject-data.md)
+- Overwrite a token - Use [Inject Data](/docs/Actions/inject-data.md)
+
+## `Related Actions`
+
+| Action Name                                                     | Description                                                                        |
+| --------------------------------------------------------------- | ---------------------------------------------------------------------------------- |
+| [Update form with AJAX](/docs/Actions/update-form-with-ajax.md) | Form module only, it updates the fields based on the values from the tokens.       |
+| [Display Message](/docs/Actions/display-message.md)             | Form Module only, shows a message after the executions of set of actions/workflow. |
+
+
+## `Input Parameter Reference`
+
+| Parameter                 | Description                                                                                                                 | Supports Variables |
+| ------------------------- | --------------------------------------------------------------------------------------------------------------------------- | ------------------ |
+| Token -> Token Expression | The tokens and/or text (expression) you want to be evaluated or reevaluated                                                 | Yes                |
+| Number of Recursions      | Specify how many times to apply tokenization. This is useful when tokens contain other tokens that also need to be replaced | Yes                |
+
+## `Output Parameters Reference`
+
+| Parameter         | Description                                                                 |
+| ----------------- | --------------------------------------------------------------------------- |
+| Token -> Store as | The variable name(s) in which the response is to be stored for further use. |
+
+## `What is recursion and when to use it`
+
+Recursion refers of how many times we should try to check for tokens in the expression and try to replace them. 
+
+If recursion is set to 1 then this action will work as [Execute Token](/docs/Actions/execute-token.md) action. Will assign the Expression to a token.
+
+There are situation ([like example number 2](#2-apply-tokens-inside-a-template-recursion-1)) where you save a template (let's say email template) and you save it with tokens instead of the First Name and Last Name so you can save only one copy and when you need to deliver that email you only have to replace the Variables. 
+
+Eg. Email Template: Hi, [User:FirstName] [User:LastName]. Please contact me!
+
+Let's say we have the above template stored in \[EmailTemplate\]
+
+When will bring the template value will have to set it's value in a token that will contain tokens itself so we will need to set the recursion to 2 to replace the token to the template and the template to a plain token. See below example for Recursion set to 2:

```

Comment:
Split in multiple sentences so it's easier to follow. Also, will = we'll and it's  = its?

Reply:
Simplified it